lib/Doctrine/Migrations/Metadata/Storage/TableMetadataStorage.php

```diff
@@ -177,6 +193,10 @@ private function needsUpdate(Table $expectedTable): ?TableDiff
 
     private function isInitialized(): bool
     {
+        if ($this->isInitialized) {
```

Comment:
shouldn't you also memoize the result of `tablesExist`, and distinguish `null` vs `false` (we know it is not initialized yet) ?